include/onnxruntime/core/framework/run_options.h

```diff
@@ -17,6 +21,13 @@ struct OrtRunOptions {
   int run_log_verbosity_level = 0;  ///< VLOG level if debug build and run_log_severity_level is 0 (VERBOSE).
   std::string run_tag;              ///< A tag for the Run() calls using this.
 
+  // An execution provider specific run option is required for some execution providers, 
+  // each of which has its own resources shared within its platform.
+  unordered_map<string, void*> provider_run_options;
+
+  // Override the default CPU allocator with a custom allocator passed from the hosting service during each run. 
+  onnxruntime::AllocatorPtr custom_cpu_allocator;
+
```

Comment:
This feels very specific to your use case. Any reason why we can't use a single completely generic 'map<string, void*>named_options' (or something like that) member? It may be for EP specific things. It may be for operator specific things (e.g. I might configure the RNN ops to use exact implementations of the activation functions instead of approximations).